lib/henkei.rb

```diff
@@ -280,4 +290,16 @@ def self.switch_for_type(type)
     end
   end
   private_class_method :switch_for_type
+
+  def self.with_timeout(seconds)
+    if seconds
+      Timeout.timeout(seconds) do
```

Comment:
Hmm not so sure about this. See https://www.mikeperham.com/2015/05/08/timeout-rubys-most-dangerous-api/ for some explanations for why (or simply do a web search for "Ruby Timeout problems")